core/iwasm/include/wasm_export.h

```diff
@@ -295,6 +300,14 @@ typedef struct wasm_global_inst_t {
     void *global_data;
 } wasm_global_inst_t;
 
+/* Table instance*/
+typedef struct wasm_table_inst_t {
+    wasm_valkind_t elem_kind;
+    uint32_t cur_size;
+    uint32_t max_size;
+    const void **elems;
```

Comment:
Is there any requirement to get the element of table? This PR creates function instances for all functions in aot runtime, which increases the footprint, in the original design, we only create aot function instances for export functions, and if to call a function in the table, we can call wasm_runtime_call_indirect. Can we just add API to the function index of a specific table element?

Reply:
OK, adding elems to wasm_table_inst_t sounds good, my concern is not to create all function instances in aot mode, and the type of elems should not be `const void **elems`, in fact, it is `uintptr_t *` when GC isn't enabled and `void **` when GC is enabled, see table_elem_type_t. So firstly, how about changing it to `void *elems` and add comment in wasm_export.h to it?

And secondly, for wasm_runtime_table_get_func_inst (suggest changing to wasm_table_get_func_inst?), we can get the function index from the element and then get the related function instance from module instance with the func index and the input module inst. Could we just get the function instance for the exported functions? e.g. return NULL for unexposed function so that in AOT we don't need to create function instance again?